bootstrap/src/main/java/jenkins/bootstrap/Plugin.java

```diff
@@ -0,0 +1,216 @@
+        return new File(archive.getParentFile(), name.substring(0,idx));
+    }
+
+    /**
+     * Returns true if this plugin overrides core components.
+     */
+    private boolean isCoreComponentOverride() throws IOException {
+        return "true".equals(manifest.getMainAttributes().getValue("Core-Component"));
+    }
+
+    public String getShortName() {
+        return manifest.getMainAttributes().getValue("Short-Name");
+    }
+
+    public String getLongName() {
+        return manifest.getMainAttributes().getValue("Long-Name");
+    }
+
+
+    public boolean isDisabled() {
+        File disableFile = new File(archive.getPath() + ".disabled");
+        return disableFile.exists();
+    }
+
+    /**
+     * Loads jar file manifest from the archive.
+     */
+    private Manifest loadManifest() throws IOException {
+        try (JarFile jar = new JarFile(archive)) {
+            ZipEntry e = jar.getEntry("META-INF/MANIFEST.MF");
+            try (InputStream is = jar.getInputStream(e)) {
+                return new Manifest(is);
+            }
+        }
+    }
+
+    /**
+     * Unpacks {@link #archive} into {@link #destDir} with uptodate check.
+     *
+     * <p>
+     * Up-to-date check needs to be consistent with {@coe ClassicPluginStrategy} in core.
+     *
+     * <p>
+     * This simplified version does not support automatic building of classes.jar from WEB-INF/classes.
+     * Pluggable core components are not supposed to have any classes, just WEB-INF/lib files.
+     */
+    private void explode() throws IOException {
+        destDir.mkdirs();
+
+        // timestamp check
+        File explodeTime = new File(destDir,".timestamp2");
+        if(explodeTime.exists() && explodeTime.lastModified()==archive.lastModified())
+            return; // no need to expand
+
+        if (destDir.exists()) {
+            // delete the contents so that old files won't interfere with new files
+            if (destDir.isDirectory())
+                Files.walk(destDir.toPath()).sorted(Comparator.reverseOrder()).map(Path::toFile).forEach(File::delete);
+            else
+                destDir.delete();
+        }
+        destDir.mkdirs();
+
+        // expand the contents
+        try (JarFile jar = new JarFile(archive)) {
+            Enumeration enumEntries = jar.entries();
+            while (enumEntries.hasMoreElements()) {
+                JarEntry file = (JarEntry) enumEntries.nextElement();
+                File f = new File(destDir + File.separator + file.getName());
+                if (file.isDirectory()) { // if its a directory, create it
+                    f.mkdir();
+                    continue;
+                }
+                try (InputStream is = jar.getInputStream(file); // get the input stream
+                    FileOutputStream fos = new FileOutputStream(f)) {
+                    while (is.available() > 0) {  // write contents of 'is' to 'fos'
```

Comment:
indentation

Reply:
nop. Two line try-with-ressources

Reply:
don't we typically double-indent the extra lines of the first statement then? Or is try-with-resources different somehow?

Reply:
I don't know if there any convention for this purpose
for my personal stuff I use to align variable definition, then use a single indent for the code block. In terms of readability all variables (`fos` in this case) appear within the block indentation level, which actually reflect their visibility

Reply:
> I use to align variable definition

Isn't the current code off by one column on the second line then?

Reply:
Yeah, would be more apparent if indented one character, that is what I was missing.